### PR TITLE
Allow functional tests to run on travis, update tests for stability.

### PR DESCRIPTION
**Pull request by rohandubal**
Add a test app host for functional tests. Update tests to use a credentials file. Update travis config to use functional and unit tests.

*Description of changes:*

- Tests to run with travis build
- Credentials are in encrypted file
- Override to run functional tests by passing inline values in .swift file
- Capture backend and test setup for functional tests

*Output of test run*

```
Test Case '-[AWSAppSyncTests.AWSAppSyncTests testSubscription_Stress]' passed (12.155 seconds).
Test Suite 'AWSAppSyncTests' passed at 2018-09-11 16:47:28.875.
	 Executed 7 tests, with 0 failures (0 unexpected) in 18.980 (18.984) seconds
Test Suite 'AppSyncMQTTClientTests' started at 2018-09-11 16:47:28.875
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testDisconnectDelegate]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testDisconnectDelegate]' passed (1.009 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testIgnoreSubscriptionsWithoutInterestedTopics]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testIgnoreSubscriptionsWithoutInterestedTopics]' passed (2.027 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testMQTTConnectionAttempt]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testMQTTConnectionAttempt]' passed (1.006 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testReceiveMessageDelegate]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testReceiveMessageDelegate]' passed (1.007 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testStopSubscriptionsOnWatcherDealloc]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testStopSubscriptionsOnWatcherDealloc]' passed (1.007 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscribeAndUnsubscribe]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscribeAndUnsubscribe]' passed (1.007 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscribeTopicsAfterConnected]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscribeTopicsAfterConnected]' passed (1.007 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscriptionsCoalescing]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscriptionsCoalescing]' passed (2.026 seconds).
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscriptionsWithInterestedTopics]' started.
Test Case '-[AWSAppSyncTests.AppSyncMQTTClientTests testSubscriptionsWithInterestedTopics]' passed (1.005 seconds).
Test Suite 'AppSyncMQTTClientTests' passed at 2018-09-11 16:47:39.982.
	 Executed 9 tests, with 0 failures (0 unexpected) in 11.101 (11.106) seconds
Test Suite 'AuthProviderTests' started at 2018-09-11 16:47:39.982
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithApiKey]' started.
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithApiKey]' passed (0.004 seconds).
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithCognitoUserPool]' started.
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithCognitoUserPool]' passed (0.004 seconds).
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithCognitoUserPoolAsync]' started.
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithCognitoUserPoolAsync]' passed (0.007 seconds).
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithOIDCToken]' started.
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithOIDCToken]' passed (0.006 seconds).
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithOIDCTokenAsync]' started.
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanAuthenticateWithOIDCTokenAsync]' passed (0.005 seconds).
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanHandleAsyncTokenErrors]' started.
2018-09-11 16:47:40.010369-0700 AWSAppSyncTestApp[53154:792690] [BoringSSL] Function nw_protocol_boringssl_handshake_negotiate_proceed: line 387 Handshake failed. Disconnecting the session
Test Case '-[AWSAppSyncTests.AuthProviderTests testCanHandleAsyncTokenErrors]' passed (0.005 seconds).
Test Suite 'AuthProviderTests' passed at 2018-09-11 16:47:40.015.
	 Executed 6 tests, with 0 failures (0 unexpected) in 0.031 (0.033) seconds
Test Suite 'AWSAppSyncTests.xctest' passed at 2018-09-11 16:47:40.016.
	 Executed 22 tests, with 0 failures (0 unexpected) in 30.112 (30.125) seconds
Test Suite 'All tests' passed at 2018-09-11 16:47:40.016.
	 Executed 22 tests, with 0 failures (0 unexpected) in 30.112 (30.126) seconds
2018-09-11 16:47:40.029740-0700 AWSAppSyncTestApp[53154:792690] TIC TCP Conn Failed [18:0x6080003654c0]: 3:61 Err(61)
2018-09-11 16:47:40.030061-0700 AWSAppSyncTestApp[53154:792690] NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, 61)
2018-09-11 16:47:40.030172-0700 AWSAppSyncTestApp[53154:792690] Task <EE4DCF1F-4373-4F04-A8D0-FAD1E2CA7F30>.<1> HTTP load failed (error code: -1200 [3:61])
2018-09-11 16:47:40.065026-0700 AWSAppSyncTestApp[53154:793956] Task <EE4DCF1F-4373-4F04-A8D0-FAD1E2CA7F30>.<1> finished with error - code: -1200
2018-09-11 16:47:40.069826-0700 AWSAppSyncTestApp[53154:792690] TIC TCP Conn Failed [19:0x60c000365b80]: 3:61 Err(61)
  1 language: swift
2018-09-11 16:47:40.070265-0700 AWSAppSyncTestApp[53154:792690] NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, 61)
2018-09-11 16:47:40.070522-0700 AWSAppSyncTestApp[53154:792690] Task <7283460B-B12C-485B-B54E-085B7CFDC27E>.<1> HTTP load failed (error code: -1200 [3:61])
2018-09-11 16:47:40.071162-0700 AWSAppSyncTestApp[53154:793227] Task <7283460B-B12C-485B-B54E-085B7CFDC27E>.<1> finished with error - code: -1200
** TEST SUCCEEDED **
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rohandubal** [comment]
Updated the README with the suggestions. Thanks!

File: README.md [hunk 1]
```diff
@@ -53,6 +53,42 @@ You can find a step by step walk through of setting up a backend and accessing i
...
+
+If you are contributing to the SDK, it is recommended to add some unit/ functional tests and evaluate against existing tests.
+
+- Unit Tests
+
+  Unit Tests do not require any specific setup and can be run directly from your Xcode IDE.
+
+- Functional Tests
+
+  For running funcitonal tests, we will need the following:
```
Comment (cbommas) [inline README.md]: typo on "functional"

File: README.md [hunk 2]
```diff
@@ -53,6 +53,42 @@ You can find a step by step walk through of setting up a backend and accessing i
...
+  - A Cognito Identity Pool with unauthenticated identities supported
+  - The Cognito Identity Pool's unauth role should have the `AppSync Invoke Full Access` permission
+  
+   You can get the backend setup by following the steps below:
+
+  - Go to [AWS AppSync console](https://console.aws.amazon.com/appsync/home).
+  - Click on `Create New API` and then select `Event App` and hit `Create`, take a note of the `API URL` of the created API.
+  - Once the creation completes, select `Settings` from left side of the console and then select `AWS Identity and Access Management (IAM)` as the authorization type.
+  - Next, create a new Cognito Identity Pool and attach `AppSync Invoke Full Access` permission to the unauth role of the Identity Pool. Keep a note of the Identity Pool ID of newly created pool.
+  - Add a file appsync_test_credentials.json (see sample below) in the `AWSAppSyncTests` folder and replace the values for `AppSyncEndpoint`,  `CognitoIdentityPoolId` and regions if required:
```
Comment (cbommas) [inline README.md]: Pull this out of the bullet list and rephrase it as follows - "Finally, you will need to setup a config file locally to access the server. Add a file ..."

File: README.md [hunk 3]
```diff
@@ -53,6 +53,42 @@ You can find a step by step walk through of setting up a backend and accessing i
...
+    ```json
+    {
+      "AppSyncEndpoint": "https://asd32hl34523454532.appsync-api.us-east-1.amazonaws.com/graphql",
+      "AppSyncRegion": "us-east-1",
+      "CognitoIdentityPoolId": "us-east-1:abcas234-1234-12324-b4b7-aaa0c0831234",
+      "CognitoIdentityPoolRegion": "us-east-1"
+    }
+    ```
+  - Now you should be able to run the functional tests!
```
Comment (cbommas) [inline README.md]: Pull this out of the bulleted list.

**Pull request merged**